game/src/js/scenes/TitleScene.js

```diff
@@ -62,11 +62,15 @@ export default class TitleScene extends Phaser.Scene {
     });
 
     if(ServiceWorker.isSupported()) {
+      const assets = this.cache.json.get('assets');
       let serviceWorker = new ServiceWorker();
       let checkbox = new Checkbox(this, 470, 500, 'ENABLE OFFLINE MODE', serviceWorker.isRegistered());
       checkbox.onPointerDown(function(obj) {
         if(obj.isChecked()) {
           serviceWorker.register();
+          setTimeout(function() {
```

Comment:
Is this to give enough time for the service worker to be registered? It would be better to return a promise from `serviceWorker.register()` that resolves when the service worker is ready.

Reply:
Yes, I should have done that in the first place.